hyperspectral/calibrate.py

```diff
@@ -14,14 +14,170 @@
 import json
 import numpy as np
 import os
-import subprocess
 import spectral.io.envi as envi
 from netCDF4 import Dataset
+from PIL import Image
+from datetime import date, datetime, timedelta
 
+from hyperspectral_calculation import pixel2Geographic, test_pixel2Geographic, solar_zenith_angle
 
-raw_root = "/home/extractor/sites/ua-mac/raw_data"
+
+raw_root = "/home/extractor/hs_calib"
 calib_root = "/home/extractor"
 
+
+def create_empty_netCDF(raw_file, out_file):
+    if not os.path.isdir(os.path.dirname(out_file)):
+        os.makedirs(os.path.dirname(out_file))
+
+    empty = Dataset(out_file, "w", format="NETCDF4")
+
+    # Get values dynamically from Header file
+    hdr_file = raw_file + ".hdr"
+    hdr_samples = None
+    hdr_lines = None
+    hdr_bands = None
+    with open(hdr_file, 'r') as hdr:
+        for l in hdr.readlines():
+            if l.startswith('samples'):
+                hdr_samples = int(l.split("=")[1].strip())
+            if l.startswith('lines'):
+                hdr_lines = int(l.split("=")[1].strip())
+            if l.startswith('bands'):
+                hdr_bands = int(l.split("=")[1].strip())
+            if hdr_samples is not None and hdr_lines is not None and hdr_bands is not None:
+                break
+
+    # Get number of frames
+    frame_file = raw_file.replace("_raw", "_frameIndex.txt")
+    with open(frame_file, 'r') as frames:
+        num_frames = len(frames.readlines()[1:])
+
+    # add dimensions
+    empty.createDimension("wavelength", hdr_bands)
+    empty.createDimension("x", hdr_lines)
+    empty.createDimension("y", hdr_samples)
+    empty.createDimension("time", num_frames)
+
+    # raw reflectance
+    # TODO: Do we need this if it is never populated?
+    v = empty.createVariable("xps_img", "u2", ("wavelength", "x", "y"))
```

Comment:
this could be dumped to a different *raw.nc file populated with the raw sensor counts and other metadata to allow re calibration. This would replace the camera outputs in netcdf that can be a) losslessly compressed with some of the nco tools and b) perhaps be efficiently done without the memory limitations (and even with parallelization if helpful).

Reply:
but then we have another thing to maintain and confuse ourselves and others with 🤷‍♂.